Deprecate `linear_norm` as a possible value for `model_confidence` (#9045)
RasaHQ/rasa: changelog/9045.removal.md

> Setting `model_confidence=linear_norm` is deprecated and will be removed in Rasa Open Source `3.0.0`.

Reviewer: ```suggestion
The option `model_confidence=linear_norm` is deprecated and will be removed in Rasa Open Source `3.0.0`.
```

> Setting `model_confidence=linear_norm` is deprecated and will be removed in Rasa Open Source `3.0.0` .

Reviewer 2: ```suggestion
Setting `model_confidence=linear_norm` is deprecated and will be removed in Rasa Open Source `3.0.0`.
```

Rasa Open Source `2.3.0` introduced `linear_norm` as a possible value for `model_confidence`
parameter in machine learning components such as `DIETClassifier`, `ResponseSelector` and `TEDPolicy`.
Based on user feedback, we have identified multiple problems with this option and hence

> using `model_confidence=linear_norm` is now deprecated and

Reviewer: Rephrasing:
"Based on user feedback, we have identified multiple problems with this option. Therefore,`model_confidence=linear_norm` is now deprecated and"

will be removed in Rasa Open Source `3.0.0` . If you were using `model_confidence=linear_norm` for any of the mentioned components,
it is recommended to revert it back to `model_confidence=softmax` and re-train the assistant. Post re-training,

> it is recommended to [tune the thresholds for fallback components](./fallback-handoff.mdx#fallbacks) as well.

Reviewer: Let's use the active voice: "it is recommended" -> "we recommend"

"Post re-training," -> "After re-training," 

"as well" <- you can omit this.

> it is recommended to [tune the thresholds for fallback components](./fallback-handoff.mdx#fallbacks) as well.

Reviewer 2: ```suggestion
it is recommended to [re-tune the thresholds for fallback components](./fallback-handoff.mdx#fallbacks) as well.
```
